perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -3469,15 +3487,23 @@ public void deleteAllAttributeAuthz(PerunSession sess, AttributeDefinition attri
 		}
 
 		try {
-			return jdbc.query("SELECT " + getAttributeMappingSelectQuery("usr_fac") + ", users.id FROM attr_names " +
-							"JOIN service_required_attrs ON attr_names.id=service_required_attrs.attr_id AND service_required_attrs.service_id=? " +
-							"JOIN users ON " + BeansUtils.prepareInSQLClause("users.id", users) +
-							"LEFT JOIN user_facility_attr_values usr_fac ON attr_names.id=usr_fac.attr_id AND facility_id=? AND user_id=users.id " +
-							"WHERE namespace IN (?,?,?)",
-					new UserAttributeExtractor(sess, this, users, facility), service.getId(), facility.getId(), AttributesManager.NS_USER_FACILITY_ATTR_DEF, AttributesManager.NS_USER_FACILITY_ATTR_OPT, AttributesManager.NS_USER_FACILITY_ATTR_VIRT);
-		} catch (EmptyResultDataAccessException ex) {
-			return new HashMap<>();
-		} catch (RuntimeException ex) {
+			return jdbc.execute("SELECT " + getAttributeMappingSelectQuery("usr_fac") + ", users.id FROM attr_names " +
+				"JOIN service_required_attrs ON attr_names.id=service_required_attrs.attr_id AND service_required_attrs.service_id=? " +
+				"JOIN users ON users.id " + Compatibility.getStructureForInClause() +
+				"LEFT JOIN user_facility_attr_values usr_fac ON attr_names.id=usr_fac.attr_id AND facility_id=? AND user_id=users.id " +
+				"WHERE namespace IN (?,?,?)", (PreparedStatementCallback<HashMap<User, List<Attribute>>>) preparedStatement -> {
+					Array sqlArray = ((PerunBlImpl)perun).getDatabaseManagerBl().prepareSQLArrayOfNumbers(users, preparedStatement);
+					preparedStatement.setInt(1, service.getId());
+					preparedStatement.setArray(2, sqlArray);
+					preparedStatement.setInt(3, facility.getId());
+					preparedStatement.setNString(4, AttributesManager.NS_USER_FACILITY_ATTR_DEF);
```

Comment:
calling setNString() will fail on PostgreSQL with SQLFeatureNotSupportedException, use setString() for all databases, the constants contain only ASCII characters